atd-vzd/migrations/migration_atd_txdot_units_2023_06_29--1509.sql

```diff
@@ -0,0 +1,6 @@
+-- trigger when atd_mode_category is changed
+CREATE OR REPLACE TRIGGER atd_txdot_units_mode_category_metadata_update
+    AFTER UPDATE ON atd_txdot_units
+    FOR EACH ROW
+    WHEN (OLD.atd_mode_category IS DISTINCT FROM NEW.atd_mode_category)
+    EXECUTE FUNCTION atd_txdot_units_mode_category_metadata_update();
```

Comment:
wondering if we need another trigger like this on inserts to the `atd_txdot_crashes` table?

Reply:
Rose - I could use some confirmation on whether I'm looking at this wrong but it appears to occur on insert and update in the `atd_txdot_crashes_updates_audit_log` function. It is buried in there and reminds me of some of the conversation about improvements we all had in one of the recent meetings. 🙏

Reply:
thanks, y'all! as far as i know it is buried [here](https://github.com/cityofaustin/atd-vz-data/blob/2a1236084f647d2c074bbf320a693cf432a73794/atd-vzd/triggers/atd_txdot_crashes_updates_audit_log.sql#L152) in a catch-all trigger that runs on insert or update to the `atd_txdot_crashes` crashes table. it was difficult for me to find initially because the trigger is labeled as an audit log trigger but does much more than that. definitely agree that making these triggers more discrete would make them much more manageable!

Reply:
@roseeichelmann i just reread your comment and it seems like maybe the trigger isn't firing when a new record is added - if that's the case i'll definitely need to do some digging to figure out why!

Reply:
@mddilley ahh yes it does look like there is a trigger on the `atd_txdot_crashes` table that executes the function `atd_txdot_crashes_updates_audit_log` where the metadata column is updated like @tillyw linked to. that `atd_txdot_crashes_updates_audit_log` function is so involved and updates so many diff things its hard to keep track of, it seems like a really good candidate to downsize/move away from using it 😅 but for now if its correctly updating the metadata column on inserts to the crashes table then ignore my comment, but looks like maybe its not?

Reply:
nvm, on a second glance at new crash records it looks like they do have the `atd_mode_category_metadata` filled out. so i think that value is being calculated on inserts like we expect, my bad!